src/store/mutations.ts

```diff
@@ -145,9 +145,10 @@ export default {
       updates: Partial<HistoryItem>;
     }
   ) {
-    const item = state.history[network]?.[walletId].find((i) => i.id === id);
-    if (item) {
-      Object.assign(item, updates);
+    const itemIndex = state.history[network]?.[walletId].findIndex((i) => i.id === id);
+    if (itemIndex != undefined && itemIndex >= 0) {
+      const item = state?.history?.[network]?.[walletId]?.[itemIndex]
+      if (item) Vue.set(state.history[network]![walletId], itemIndex, Object.assign(Object.assign({}, item), updates))
```

Comment:
isn't that object assign same as doing
`Object.assign({}, item, updates)` ? This seems more readable